tools/codegen/gen_functionalization_type.py

```diff
@@ -0,0 +1,333 @@
+from tools.codegen.api import cpp
+from tools.codegen.api import dispatcher
+from tools.codegen.api.types import DispatcherSignature, CppSignatureGroup
+from tools.codegen.context import method_with_native_function
+from tools.codegen.model import (
+    Argument, NativeFunction, NativeFunctionsGroup, Return, SchemaKind, BackendIndex, DispatchKey
+)
+from typing import List, Optional, Union, Tuple, Dict
+from dataclasses import dataclass
+from typing_extensions import Literal
+from tools.codegen.utils import mapMaybe, Target, assert_never
+
+from tools.autograd.gen_inplace_or_view_type import (
+    gen_formals, get_view_info, modifies_arguments, COMPOSITE_VIEW_FULL_NAMES
```

Comment:
It looks like the bazel build expects all of the `tools/autograd` codegen to depend on all of the `tools/codegen` codegen. I might move these functions into this file to fix it.